File: DonRyu/2022-10-17-Jump Game.js

var canJump = function(nums) {
  var len = nums.length;
  var max = 0;
  for (var i = 0; i < len; i++) {
    if (i > max) return false;
    max = Math.max(max, i + nums[i]);
  }
  return true;
};

File: Chris/2022-10-17-Jump Game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        
        
        j = 0 # farthest reacheable index 
        
        for i in range(len(nums)-1):
            j = max(i + nums[i], j)
            
            if j >= len(nums)-1:
                return True
            
            if j <= i:
                return False
            
            
        
        return True
            
            
        